hazelcast/src/main/java/com/hazelcast/map/impl/operation/PartitionCheckIfLoadedOperation.java

```diff
@@ -72,7 +72,8 @@ public Object getResponse() {
 
     @Override
     public boolean returnsResponse() {
-        return !waitForKeyLoad;
+        // has to return true in all cases in order to enable the default retry invocation mechanism
+        return true;
```

Comment:
Sure I noticed it too, but I wanted to have a comment, since it's a problematic part that may cause a deadlock. It's a part of a future PRD, so I would prefer to keep it.